src/meta/src/hummock/manager/mod.rs

```diff
@@ -726,7 +726,9 @@ where
         Ok(Some(compact_task))
     }
 
-    pub async fn cancel_compact_task(&self, compact_task: &CompactTask) -> Result<bool> {
+    /// Cancels a compaction task no matter it's assigned or unassigned.
+    pub async fn cancel_compact_task(&self, compact_task: &mut CompactTask) -> Result<bool> {
+        compact_task.task_status = false;
```

Comment:
Some unimportant points, `task_status is a bool` This point is a bit confusing to me, can rename it to `alive` to identify whether the task is alive or not. If I am incorrect please ignore it

Reply:
Let's leave the renaming issue to other PR :)